### DISCUSSION TITLE
APNs will no longer support the legacy binary protocol after March 31, 2021

### DISCUSSION THREAD

**Discussion by puremana** [kreait/firebase-php#580] [Q&A] [answered]
Hi, I have just gotten an email stating

```The HTTP/2 based Apple Push Notification service (APNs) provider API lets you take advantage of great features, such as authentication with a JSON Web Token, improved error messaging, and per notification feedback. If you still send push notifications with the legacy binary protocol, make sure to upgrade to the APNs provider API as soon as possible. APNs will no longer support the legacy binary protocol after March 31, 2021. ```

We currently use this library for sending notifications. Just checking this isn't going to cause any issues? I'm not sure if you are sending this to only Firebase or directly to Apple.

Thanks for all your work on this library!

**Comment by jeromegamez** [comment, marked as answer]
That's a good question! 

Firebase (the service) handles the actual sending of the messages. We're just sending a "cloud message" configuration to Firebase, and Firebase then translates it into messages targeted at the specific platform, so I'm confident that this shouldn't be an issue with this (or any other) Firebase SDK. 🤞

**Reply by puremana** [reply]
Awesome! Thanks for the quick reply.
